include/bluetooth/mesh/gen_lvl.h

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/**
+ * @file
+ * @defgroup bt_mesh_gen_lvl Bluetooth Mesh Generic Level models
+ * @{
+ * @brief Common API for the Bluetooth Mesh Generic Level models.
+ */
+#ifndef BT_MESH_GEN_LVL_H__
+#define BT_MESH_GEN_LVL_H__
+
+#include <bluetooth/mesh.h>
+#include <bluetooth/mesh/model_types.h>
+
+/** @cond INTERNAL_HIDDEN  */
+#define GEN_LVL_OP_GET			BT_MESH_MODEL_OP_2(0x82, 0x05)
+#define GEN_LVL_OP_SET			BT_MESH_MODEL_OP_2(0x82, 0x06)
+#define GEN_LVL_OP_SET_UNACK		BT_MESH_MODEL_OP_2(0x82, 0x07)
+#define GEN_LVL_OP_STATUS		BT_MESH_MODEL_OP_2(0x82, 0x08)
+#define GEN_LVL_OP_DELTA_SET		BT_MESH_MODEL_OP_2(0x82, 0x09)
+#define GEN_LVL_OP_DELTA_SET_UNACK	BT_MESH_MODEL_OP_2(0x82, 0x0A)
+#define GEN_LVL_OP_MOVE_SET		BT_MESH_MODEL_OP_2(0x82, 0x0B)
+#define GEN_LVL_OP_MOVE_SET_UNACK	BT_MESH_MODEL_OP_2(0x82, 0x0C)
+/** @endcond */
+
+/** Generic Level set message parameters.  */
+struct gen_lvl_set {
+	/** New level. */
+	s16_t level;
+	/** Whether this is a new action. */
+	bool new_transaction;
+	/**
+	 * Transition time parameters for the state change. Setting the
+	 * transition to NULL makes the server use its default transition time
+	 * parameters.
+	 */
+	const struct model_transition *transition;
+};
+
+/** Generic Level delta set message parameters.  */
+struct gen_lvl_delta_set {
+	/** Translation from original value. */
+	s32_t delta;
+	/**
+	 * Whether this is a new transaction. If true, the delta should be
+	 * relative to the current value. If false, the delta should be
+	 * relative to the original value in the previous delta_set command.
+	 */
+	bool new_transaction;
+	/**
+	 * Transition time parameters for the state change. Setting the
+	 * transition to NULL makes the server use its default transition time
+	 * parameters.
+	 */
+	const struct model_transition *transition;
+};
+
+/** Generic Level move set message parameters. */
+struct gen_lvl_move_set {
```

Comment:
This definition is equal to `struct gen_lvl_set` could it be re-used?

Reply:
They sort of accidentally have the same field types, but `level` in set and `delta` in this one are conceptually different values with different units. I think they should be kept different to maintain type safety and communicate intent in the callbacks.